src/main/java/edu/studyup/map/Lookup.java

```diff
@@ -37,13 +37,15 @@ private static Location getLocation(JSONObject location) {
 		return new Location(lat, lon, bounds);
 	}
 
+	//FIXED BUG #1: Reliance on default encoding
+	//SOLUTION: add "UTF-8" as parameter for InputStreamReader
 	private static JSONArray queryURL(String query) {
 		JSONArray results = new JSONArray();
 		try {
 			String urlString = "https://nominatim.openstreetmap.org/search?q=" + URLEncoder.encode(query, "UTF-8")
 					+ "&format=json";
 			URL url = new URL(urlString);
-			try (BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream()))) {
+			try (BufferedReader in = new BufferedReader(new InputStreamReader(url.openStream(), "UTF-8"))) {
```

Comment:
Good one! Wooooow, it took me 2 days to look around on StackOverflow for possible solutions but I have not been able to. I am so glad that we have code review in this assignment.